src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+import argparse
+import platform
+import sys
+import warnings
+from pathlib import Path
+from typing import Set
+
+from tox.execute.api import Execute
+from tox.execute.local_sub_process import LocalSubProcessExecuteInstance
+from tox.plugin import impl
+from tox.tox_env.python.api import PythonInfo
+from tox.tox_env.python.runner import PythonRun
+
+
+@impl
+def tox_add_option(parser):
+    parser.add_argument(
+        "--current-env",
+        action="store_true",
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-only",
+        action="store_true",
+        dest="print_deps_only",
+        default=False,
+        help="Deprecated, equivalent to `--print-deps-to -`",
+    )
+    parser.add_argument(
+        "--print-deps-to",
+        "--print-deps-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the dependencies to the given file "
+        + "(use `-` for stdout)",
+    )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the  names of the required extras to the given file "
+        + "(use `-` for stdout)",
+    )
+
+
+@impl
+def tox_register_tox_env(register):
+    register.add_run_env(CurrentEnv)
+    register.add_run_env(PrintEnv)
+
+
+@impl
+def tox_add_core_config(core_conf, config):
+    if config.options.current_env:
+        config.options.default_runner = "current-env"
+        return
+
+    if config.options.print_deps_only:
+        warnings.warn(
+            "--print-deps-only is deprecated; use `--print-deps-to -`",
+            DeprecationWarning,
+        )
+        if not config.options.print_deps_to:
+            config.options.print_deps_to = sys.stdout
+        else:
+            raise RuntimeError(
+                "--print-deps-only cannot be used together " "with --print-deps-to"
```

Comment:
you can use mutually exclusive flags with argparse, no need for this awkward dance here

Reply:
This will be also removed with the deprecated option.